### DISCUSSION TITLE
Can't calibrate mag Matek M8Q 5883

### DISCUSSION THREAD

**Discussion by MaximMalkov** [iNavFlight/inav#7280] [Q&A] [answered]
I have a problem with calibration of magnitometer Mateksys M8Q 5883 in INAV 3.0. Currently it is CW90FLIP and it finds North properly. But the problem, that I also need to roll it for 21 degrees, as it mounted with the angle. The command align_mag_roll = - 210 does not help, the magnitometer loses North after it.  

Mount for the GPS + magnitometer is the following https://www.thingiverse.com/thing:2973323

My current dump is the following:
[INAV_cli_Max_Drone_20210718_010938.txt](https://github.com/iNavFlight/inav/files/6835362/INAV_cli_Max_Drone_20210718_010938.txt)

Could anyone help with this issue

**Comment by avsaase** [comment, marked as answer]
You cannot use the alignment presets together with the manual alignment settings. Try
```
set align_mag_roll = 210
set align_mag_pitch = 1800
set align_mag_yaw = 900
```

The last two settings are equivalent to `CW90FLIP`.

**Reply by MaximMalkov** [reply]
@avsaase thanks a lot! How can I set alignment presets do DEFAULT in INAV 3.0 (in order to use manual alignment settings)? If I would set align_mag = DEFAULT, it will be CW270FLIP. Or should I set set align_mag = CW0?

I made it like this right now, is it right?

set align_mag = CW270FLIP (probably I should do it =CW0)
set align_mag_roll = 210
set align_mag_pitch = 1800
set align_mag_yaw = 900

The Quad finds North correctly right now, but it is rotating a bit in setup.

**Reply by avsaase** [reply]
Iirc setting any of the `align_mag_[axis]` settings to a non-zero value makes it ignore `align_mag`. If that is not true then set it to `DEFAULT`.

**Reply by MaximMalkov** [reply]
Thank you!
